### ISSUE TITLE
name choice of `maxLinkDim`

### ISSUE THREAD

**Issue by orialb** [ITensor/ITensors.jl#79]
Hi,
first let me say that I am really excited about iTensor coming to Julia. I used the C++ version a couple of years ago and this new port looks really promising. thanks for your work!

I was wondering regarding the naming of `maxLinkDim`, for two reasons:
1. According to the Julia style guide functions should be in lowercase, so it seems that `maxlinkdim` would be more julian (this is also true for `maxDim`).
2. Wouldn't calling it `maxbonddim` be more in line with the tensor networks literature ? I never saw it referred to as "link dimension" (but I am aware the core authors of the library have much greater expertise in tensor networks than me, so please correct me if I'm wrong). I guess this is related to the fact that the bond indices of the MPS are tagged as "Link".

Probably point 2 is less important, although it might make it more intuitive and discoverable for new users who are familiar with the literature.

**Comment by mtfishman** [comment]
Hi,

Thanks for your comments, contributions and interest in ITensor.

1. Miles and I went back and forth about using camel casing like we do in the C++ version versus using the all lower case style of Julia. We both felt like camel casing is more readable, so we were trying to decide if following along with the Julia convention for the sake of being consistent with Julia Base and other Julia packages was worth it. In the end, we decided that we will continue with the C++ convention, since we prefer how it looks and reads and there didn't seem to be much of a downside (in general, it is pretty clear when someone is using an ITensor function, and we can follow the Julia convention for Base functions that we overload on ITensors which makes it clear it is an overloaded function). I think once people realize that camel casing is the convention that ITensors.jl uses, they will get used to it and it won't lead to too much trouble.
Right now the convention in ITensors.jl is mixed, just based on when the functions were written and who wrote them, but we plan to change the names to camel casing and use camel casing going forward.

2. Part of the goal of ITensor going forward is to be of interest to people not just in the physics/tensor network community (for example, we hope it will be of interest in data science/machine learning). From that perspective, I think the term "bond" is fairly jargony (I'm not sure if it is used outside of the tensor network community). I think "link" is a bit more neutral, and I don't know if there is any standard term that means "indices/dimensions summed over in a series of tensor contractions". I would say that it has worked pretty well so far, and I don't see a very strong reason to switch right now (once you learn what we use "link" for, we use it consistently throughout the library so I think it is easy enough to remember).

Cheers,
Matt
